config/routes.rb

```diff
@@ -43,4 +40,20 @@
       resources :group_assignments, path: 'group-assignments'
     end
   end
+
+  namespace :stafftools do
+    mount Sidekiq::Web  => '/sidekiq'
```

Comment:
Just went through this same process in education-web. The solution we did was to extend `ActionDispatch::Request` and delegate to in in `ApplicationController`. Very nice for removing duplication and authorizing once in both Route Constraints and Controller

app/request/user_session_methods.rb
```ruby
module UserSessionMethods
  def current_user
    @current_user ||= User.find_by_id session[:user_id]
  end

  def current_user=(user)
    @current_user = user
    session[:user_id] = user.id
    session[:valid_until] = 1.days.from_now
  end

  def signed_in?
    !!(current_user && session[:valid_until] &&
      session[:valid_until] > Time.now)
  end
  ...
end

ActiveSupport.on_load(:action_controller) do
  ActionDispatch::Request.send(:include, UserSessionMethods)
end
```

app/controllers/application_controller
```ruby
class ApplicationController < ActionController::Base
  ...
  delegate :current_user, :current_user=, :signed_in?, :admin?,
           to: :request
  helper_method :current_user, :signed_in?, :admin?
  helper_method :admin_discount_request_path
  ...
```

```ruby
  mount Sidekiq::Web  => '/sidekiq', constraints: { admin?: true }
```